test/test_gru_cell.h

```diff
@@ -0,0 +1,60 @@
+/*
+    Copyright (c) 2013, Taiga Nomi and the respective contributors
+    All rights reserved.
+
+    Use of this source code is governed by a BSD-style license that can be found
+    in the LICENSE file.
+*/
+#pragma once
+
+#include <gtest/gtest.h>
+
+#include <functional>
+#include <vector>
+
+#include "test/testhelper.h"
+#include "tiny_dnn/tiny_dnn.h"
+
+namespace tiny_dnn {
```

Comment:
a test just with a forward pass would help also to understand how to use the api

Reply:
also, I think there should be a `setup` test which clearly highlights the attributes. Similar tests are written for conv layer and other too.

Reply:
I agree, for every individual thing of layer. I think this should be done in another PR though, this one is big enough. Should I open an issue or something like that to define the task?